prov/efa/src/efa_domain.c

```diff
@@ -287,19 +286,26 @@ int efa_domain_open(struct fid_fabric *fabric_fid, struct fi_info *info,
 
 	domain->cache = NULL;
 
-	if (enable_cache && efa_check_fork_enabled(*domain_fid)) {
+	/*
+	 * Check whether fork support is enabled when app does not request
+	 * FI_MR_LOCAL even if the cache is disabled.
+	 */
+	if (!app_mr_local && efa_check_fork_enabled(*domain_fid)) {
```

Comment:
Should this actually be `if (!app_mr_local && efa_mr_cache_enable && efa_check_fork_enabled()) {...}`?  Not that we would recommend disabling the cache (given the bugs found), but in consistency for future cleanups?

Reply:
That will allow users to proceed without `FI_MR_LOCAL`, no cache, and fork support so I don't think that's what we want here.

Reply:
I guess I question why the conditional is different than the one directly below it?

Reply:
I also don't understand, shouldn't we check efa_mr_cache_enable here?

For example, this will fail if FI_MR_LOCAL is not set but the user specifies they don't want the MR cache.

Reply:
This condition is testing for an app that has not requested FI_MR_LOCAL and has fork support enabled. It's my understanding that we are going to not allow this regardless of whether the cache is enabled or not.

Once we've made it through the fork support check the second condition will decide to enable the cache or not, based on FI_MR_LOCAL and the cache environment variable.

Reply:
I find it hard to see why we would want to fail if the app has specifically disabled the cache. It works perfectly fine and there's no error case or performance degradation, so we just work in less cases?